src/components/details/details.ts

```diff
@@ -174,7 +174,6 @@ export default class SlDetails extends ShoelaceElement {
           part="header"
           id="header"
           class="details__header"
-          role="button"
```

Comment:
We actually want this to remain `role="button"` so it announces the same way `<details>` elements do. Let's leave the role intact and change the tag from `<header>` to `<div>` instead.